OpenRA.Mods.Common/AI/SupportPowerDecision.cs

```diff
@@ -193,8 +193,8 @@ public int GetAttractiveness(FrozenActor fa, Stance stance, Player firedBy)
 							return (valueInfo != null) ? valueInfo.Cost * Attractiveness : 0;
 
 						case DecisionMetric.Health:
-							var healthInfo = fa.Info.TraitInfoOrDefault<HealthInfo>();
-							return (healthInfo != null) ? fa.HP * Attractiveness / healthInfo.HP : 0;
+							var health = fa.Actor.TraitOrDefault<IHealth>();
+							return (health != null) ? fa.HP * Attractiveness / health.HP : 0;
```

Comment:
This looks wrong (fa.HP and health.HP are the same).
It should be `fa.HP * Attractiveness / health.MaxHP`.

Reply:
It would be better to keep this querying the Info (there is generally no guarantee that an actor associated with a FrozenActor is still alive, so this pattern should be avoided).

Reply:
IHealthInfo does not provide the given information, so using this here wont work properly. Im not quite sure how to correctly handle this here?

Reply:
I presume you can extend IHealthInfo to expose MaxHP.